libvips/foreign/webpsave.c

```diff
@@ -109,15 +155,639 @@ typedef struct _VipsForeignSaveWebp {
 
 	/* Profile to embed.
 	 */
-	char *profile;
+	const char *profile;
+
+	WebPConfig config;
+
+	/* Output is written here. We can only support memory write, since we
+	 * handle metadata.
+	 */
+	WebPMemoryWriter memory_writer;
+
+	/* Write animated webp here.
+	*/
+	WebPAnimEncoder *enc;
+
+	/* Add metadata with this.
+	 */
+	WebPMux *mux;
+
+	/* The current frame coming from libvips, and the y position
+	 * in the input image.
+	 */
+	VipsRegion *frame;
+	int write_y;
 
-} VipsForeignSaveWebp;
+	/* VipsRegion is not always contiguious, but we need contiguous RGB(A)
+	 * for libwebp. We need to copy each frame to a local buffer.
+	 */
+	VipsPel *frame_bytes;
+} VipsForeignSaveWebP;
 
-typedef VipsForeignSaveClass VipsForeignSaveWebpClass;
+typedef VipsForeignSaveClass VipsForeignSaveWebPClass;
 
-G_DEFINE_ABSTRACT_TYPE( VipsForeignSaveWebp, vips_foreign_save_webp, 
+G_DEFINE_ABSTRACT_TYPE( VipsForeignSaveWebP, vips_foreign_save_webp,
 	VIPS_TYPE_FOREIGN_SAVE );
 
+static void
+vips_webp_write_unset( VipsForeignSaveWebP *write )
```

Comment:
This is a gobject method, so it ought to be `vips_foreign_save_webp_unset` (yuk).